command/agent/consul/service_client.go

```diff
@@ -201,13 +201,27 @@ func different(wanted *api.AgentServiceRegistration, existing *api.AgentService,
 		!connectSidecarDifferent(wanted, sidecar))
 }
 
+func tagsDifferent(a, b []string) bool {
+	if len(a) != len(b) {
+		return true
+	}
+
+	for i, valueA := range a {
```

Comment:
this returns *different* if the same tags, in a different order. 

there's an argument that we should consider the canonical (sorted) form, allowing users to changed the order without re-registering (which may be the case, for example, if the tags are procedurally generated in a way that doesn't guarantee consistent ordering). 
there's another argument that says if they change the order of the tags on a job update, we'll reregister the service (with the tags in exactly the order they specified).

Reply:
Consul itself persists tags in the order given, and at least for now I think Nomad should should maintain what the user provides.